src/service/expenses.service.js

```diff
@@ -0,0 +1,129 @@
+// eslint-disable-next-line no-unused-vars
+const { Expenses } = require('../model/expenses.model');
+
+const { getUserById } = require('./user.service');
+
+/**
+ * @type {Expenses[]}
+ */
+let expenses = [];
+
+const getExpenses = (expensesURLParams) => {
+  const { userId, categories, from, to } = expensesURLParams;
+
+  let filteredExpenses = [...expenses];
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.userId === Number(userId),
+    );
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.category === categories,
+    );
+  }
+
+  if (from) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.spentAt >= from,
+    );
+  }
+
+  if (to) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.spentAt <= to,
+    );
+  }
+
+  return filteredExpenses;
```

Comment:
All these _if_ approaches work, but could become something hard to maintain and scale in case of more and more properties/queries needed.
You could try storing filtering functions in a hashMap (Object works), and using the given function accordingly.
This will keep all scalability in a single place and remove multiple _if_ checks. (because your code checks all ifs, even when already passing the first one).
If you do so, remember to add a default filtering function